TaskWithSmartPointers/MyUniquePtr.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include <iostream>
+
+template <class T>
+class MyUniquePtr
+{
+public:
+	T* get() const noexcept
+	{
+		return m_ptr;
+	}
+
+	T* release() noexcept
+	{
+		T* p = m_ptr;
+		m_ptr = nullptr;
+		return p;
+	}
+
+	void reset(T* p) noexcept
+	{
+		delete m_ptr;
+		m_ptr = p;
+	}
+
+	T& operator*()
+	{
+		return *(this->m_ptr);
+	}
+
+	MyUniquePtr() // default constructor
+	{
+		m_ptr;
+	}
+
+	MyUniquePtr(T* ptr)
+	{
+		m_ptr = ptr;
+	}
+
+	MyUniquePtr(const MyUniquePtr& obj) = delete; // copy constructor is deleted
+
+	MyUniquePtr& operator=(const MyUniquePtr& obj) = delete; // copy assignment is deleted
+
+	MyUniquePtr<T>& operator=(MyUniquePtr<T>&& obj) noexcept // move assignment
+	{
+		this->m_ptr = obj.m_ptr;
+		obj.m_ptr = nullptr;
+		return *this;
+	}
+
+	MyUniquePtr(MyUniquePtr&& obj) // move constructor
+	{
+		*this = obj;
+		obj.m_ptr = nullptr;
```

Comment:
Won't obj.m_ptr be set tp nullptr by operator=? Looks like redundancy.

Reply:
Fixed